artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTSubscriptionManager.java

```diff
@@ -194,12 +191,13 @@ private synchronized void removeSubscription(String address) throws Exception {
       SimpleString internalQueueName = getQueueNameForTopic(internalAddress);
       session.getSessionState().removeSubscription(address);
 
-
-      ServerConsumer consumer = consumers.get(address);
+      Set<Consumer> queueConsumers = session.getServer().queueConsumersQuery(internalQueueName);
```

Comment:
I think this is wrong. You should only check for consumers on this Session.

Your logic is closing every consumer to the same address. You keep multiple connections and you have a failure.